src/es/olympusscanlation/src/eu/kanade/tachiyomi/extension/es/olympusscanlation/OlympusScanlation.kt

```diff
@@ -1,41 +1,81 @@
 package eu.kanade.tachiyomi.extension.es.olympusscanlation
 
+import android.app.Application
+import android.content.SharedPreferences
+import android.widget.Toast
+import androidx.preference.CheckBoxPreference
+import androidx.preference.EditTextPreference
+import androidx.preference.PreferenceScreen
 import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
+import eu.kanade.tachiyomi.source.ConfigurableSource
 import eu.kanade.tachiyomi.source.model.Filter
 import eu.kanade.tachiyomi.source.model.FilterList
 import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
 import okhttp3.HttpUrl.Companion.toHttpUrl
 import okhttp3.Request
 import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 import uy.kohesive.injekt.injectLazy
 import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.TimeZone
 import kotlin.concurrent.thread
 
-class OlympusScanlation : HttpSource() {
+class OlympusScanlation : HttpSource(), ConfigurableSource {
 
     override val versionId = 2
 
-    override val baseUrl: String = "https://olympuscomic.com"
-    private val apiBaseUrl: String = "https://dashboard.olympuscomic.com"
+    override val baseUrl: String get() = when {
+        System.getenv("CI") == "true" -> defaultBaseUrl
+        else -> preferences.getPrefBaseUrl()
```

Comment:
calling the pref each time baseurl is accessed? consider making another var with custom get and set to cache pref values